src/Common/Admin/Entities/Link.php

```diff
@@ -33,19 +33,33 @@ class Link extends Base_Entity {
 	 */
 	private string $text = '';
 
+	/**
+	 * Content for the link if not a string.
+	 *
+	 * @var ?Base_Entity
+	 */
+	private ?Base_Entity $content = null;
+
 	/**
 	 * Link constructor.
 	 *
 	 * @since 6.1.0
 	 *
-	 * @param string      $url        The URL for the link.
-	 * @param string      $text       The text for the link.
-	 * @param ?Classes    $classes    The classes for the link.
-	 * @param ?Attributes $attributes The attributes for the link.
+	 * @param string                  $url        The URL for the link.
+	 * @param string|Base_Entity      $content    The text or entity for the link.
+	 * @param ?Classes                $classes    The classes for the link.
+	 * @param ?Attributes             $attributes The attributes for the link.
 	 */
-	public function __construct( string $url, string $text, ?Classes $classes = null, ?Attributes $attributes = null ) {
+	public function __construct( string $url, $content, ?Classes $classes = null, ?Attributes $attributes = null ) {
```

Comment:
For sure, but that is why I picked `Base_Entity` which is extended by `Container` so you can pick an individual one or the container, since the `render` from `Container` is still used in `__toString()` allowing me to not care if it's one or many.


```
// Opt 1
new Link( 'url', 'Text' );

// Opt 2
new Link( 'url', new Image( 'url' ... ) );

// Opt 3
$container = new Container( ... )->add_children( new Image( ... ), new Image( ... ) );
new Link( 'url', $container );
```